docs/network-iot/run-an-lns/buy-an-oui.mdx

```diff
@@ -0,0 +1,218 @@
+```
+
+In order to compile the copied code, the Rust compiler must be installed.
+
+```bash title="Install the Rust toolchain."
+curl https://sh.rustup.rs -sSf | sh
+```
+
+```bash title="Add Rust to path."
+source "$HOME/.cargo/env"
+```
+
+Add the protobuf library which will be used during the compile process.
+
+```bash title="Install protobuf"
+brew update && brew install protobuf
+```
+
+```bash title="Enter the directory and compile the CLI"
+cd helium-config-service-cli && cargo build --release
+```
+
+The built binary file `helium-config-service-cli` can be found under `target/release/` inside the
+`helium-config-service-cli` directory.
+
+Now that we have the CLI binary, let's put a copy in the local bin folder for simplicity's sake.
+Just copy `helium-config-service-cli` into `/usr/local/bin`. Files added to the bin folder will
+automatically be available to your command line.
+
+```bash
+sudo cp target/release/helium-config-service-cli /usr/local/bin/
+```
+
+### Generate OUI Keys
+
+Generate the keypairs you'll be using to authenticate your interactions with the config service.
+These are keys are implemented such that only you can create/modify/delete routes.  
+Run this command twice. First for the owner keypair and second for the delegate kepair. Name files
+accordingly.
+
+**Delegate Keypair**: The delegate key is actively used to create and manage routes. In case it is
+ever compromised, it can be updated using the owner key.
+
+**Owner Keypair**: The owner key is irreplaceable and should be kept safe, secure, and private at
+all times. If lost or compromised, you could lose your OUI.
+
+```bash title="Generate keypair"
+helium-config-service-cli env generate-keypair ~/<key file name>.bin
+```
+
+Run `helium-config-service-cli env generate-keypair <key file name>.bin` and you should find a file
+called `<key file name>.bin`.
+
+Print out the public key of your keypair by running:
+
+```bash
+$ helium-config-service-cli env info --keypair <key file name>.bin
+```
+
+```json {9} title="Example response from the info command highlighting the keypair."
+{
+  "arguments": {
+    "config_host": "unset",
+    "key_type_from_keypair": "ed25519",
+    "keypair": "/Users/me/test.bin",
+    "max_copies": null,
+    "net_id": null,
+    "oui": null,
+    "public_key_from_keypair": "13eY9uE5fjMzKC7PZ1nsDigiyurhXhV3bbj8pxbD2KtQTSC1ZHP"
+  },
+  {/* env vars will be discussed later */}
+}
+```
+
+The .bin files should be kept safe and never shared. The public key of these keypairs is the only
+part that should be shared.
+
+Now, please send an email to [hello@helium.foundation](mailto:hello@helium.foundation) to purchase
+an OUI and devAddrs containing the following informations.
+
+- The public key of your owner key pair
+- The public key of your delegate key pair
+- The Helium wallet address
+- The number of devAddrs you want
+
+Once you receive your OUI, the next step is to fund it. Learn how [here](fund-an-oui).
+
```

Comment:
What I am missing here is the explanation of the escrow account as that is the only account that you can lookup on blockchain and where all DC transaction can be found.
Currently the only method of retrieving the escrow account that I know of is via the stripe dc-portal .. would be good to have some documentation of how this works and how you can retrieve this based on your oui